### PR TITLE
Add python3 support

### PR DESCRIPTION
**Pull request by charliewolf**

### CODE REVIEW

File: tasktiger/_internal.py [hunk 1]
```diff
@@ -28,7 +30,7 @@ def gen_id():
     """
     Generates and returns a random hex-encoded 256-bit unique ID.
     """
-    return os.urandom(32).encode('hex')
+    return binascii.b2a_hex(os.urandom(32)).decode('utf8')
```
Comment (wojcikstefan) [inline tasktiger/_internal.py]: What's the advantage here?

File: tasktiger/task.py [hunk 2]
```diff
@@ -228,7 +229,7 @@ def _move(self, from_state=None, to_state=None, when=None, mode=None):
         try:
             pipeline.execute()
         except redis.ResponseError as e:
-            if '<FAIL_IF_NOT_IN_ZSET>' in unicode(e):
+            if '<FAIL_IF_NOT_IN_ZSET>' in text_type(e):
```
Comment (wojcikstefan) [inline tasktiger/task.py]: Does this work if the exception message contains unicode chars?

Comment (charliewolf) [inline tasktiger/task.py]: @wojcikstefan Yes, as text_type == unicode on python2 and str (which is the same as unicode) in python3

File: tasktiger/_internal.py (hunk 1 above)
Comment (charliewolf) [inline tasktiger/_internal.py]: The advantage is that it works... encode('hex') is not available in python3

**Comment by thomasst** [comment]
Thanks! Will review once I find the time.

**Comment by philfreo** [comment]
We should add `tox` support for easy testing on multiple versions

**Comment by charliewolf** [comment]
@philfreo I added it!

**Comment by charliewolf** [comment]
Seems to have some errors on py3.4 (circle ci's latest) compared to py3.5 (what I'm running). Need to investigate these.

**Comment by charliewolf** [comment]
@thomasst I fixed the py3.4 issues.

File: tasktiger/_internal.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
+import binascii
 import calendar
 import datetime
 import importlib
 import hashlib
 import json
 import os
+import six
```
Comment (thomasst) [inline tasktiger/_internal.py]: Why are we importing `six` in this file?

File: tasktiger/worker.py [hunk 4]
```diff
@@ -131,7 +132,7 @@ def _update_queue_set(self, timeout=None):
             r, w, x = select.select([fileno], [], [],
                                     0 if self._queue_set else timeout)
             if fileno in r: # or not self._queue_set:
-                message = gen.next()
+                message = six.next(gen)
```
Comment (thomasst) [inline tasktiger/worker.py]: We can just use `next(gen)`, right? I don't care about support for Python older than 2.7.

File: tasktiger/task.py (hunk 2 above)
Comment (thomasst) [inline tasktiger/task.py]: `if '<FAIL_IF_NOT_IN_ZSET>' in e.args[0]:` should work, too, and wouldn't require `six`.

File: requirements.txt [hunk 5]
```diff
@@ -1,3 +1,4 @@
 click==4.0
 redis==2.10.3
 structlog==15.1.0
+six==1.10.0
```
Comment (thomasst) [inline requirements.txt]: I propose we don't use `six`. See my other comments for reasons.

Comment (charliewolf) [inline requirements.txt]: @thomasst done.

**Comment by thomasst** [comment]
Thanks, this looks good. Will do some more testing and then merge it if everything looks good.

**Pull request merged**